- [Tophat](https://ccb.jhu.edu/software/tophat/index.shtml): First widely used spliced-aware read mapper. Current version is Tophat2

- [HISAT](http://ccb.jhu.edu/software/hisat2/index.shtml): Successor to Tophat2. Similar functionality, but n=much more computationally efficient

- [STAR](https://github.com/alexdobin/STAR): Much faster than Tophat, but requires a large amount of memort (30 Gb for human genome)

TopHat [Trapnell C, Pachter L, Salzberg SL. TopHat: discovering splice junctions with
RNA-Seq. Bioinformatics. 2009;25(9):1105–11.]
TopHat2 [Kim D, Pertea G, Trapnell C, Pimentel H, Kelley R, et al. TopHat2: accurate
alignment of transcriptomes in the presence of insertions, deletions and
gene fusions. Genome Biol. 2013;14(4):R36.]
SpliceMap [Au KF, Jiang H, Lin L, Xing Y, Wong WH. Detection of splice junctions
from paired-end RNA-seq data by SpliceMap. Nucleic Acids Res.
2010;38(14):4570–8.]
GSNAP [Wu TD, Nacu S. Fast and SNP-tolerant detection of complex variants and
splicing in short reads. Bioinformatics. 2010;26(7):873–81.]

