# de-novo transcript assembly
-Kim, C. S., Winn, M. D., Sachdeva, V., & Jordan, K. E. (2017). K-mer clustering algorithm using a MapReduce framework: application to the parallelization of the Inchworm module of Trinity. [BMC Bioinformatics, 18(1), 467](http://doi.org/10.1186/s12859-017-1881-8)

BinPacker [Liu J, Li G, Chang Z, Yu T, Liu B, et al. BinPacker: packing-based de novo
transcriptome assembly from RNA-seq data. PLoS Comput Biol.
2016;12(2):e1004772.]
Bridger [Chang Z, Li G, Liu J, Zhang Y, Ashby C, et al. Bridger: a new framework for de
novo transcriptome assembly using RNA-seq data. Genome Biol. 2015;16:30.]
Trinity [Grabherr MG, Haas BJ, Yassour M, Levin JZ, Thompson DA, et al. Full-length
transcriptome assembly from RNA-Seq data without a reference genome.
Nat Biotechnol. 2011;29(7):644–52.]
ABySS [Birol I, Jackman SD, Nielsen CB, Qian JQ, Varhol R, et al. De novo
transcriptome assembly with ABySS. Bioinformatics. 2009;25(21):2872–7.]
SOAPdenovo-Trans [Li RQ, Yu C, Li YR, Lam TW, Yiu SM, et al. SOAP2: an improved ultrafast tool
for short read alignment. Bioinformatics. 2009;25(15):1966–7.]
Oases [Schulz MH, Zerbino DR, Vingron M, Birney E. Oases: robust de novo RNA-
seq assembly across the dynamic range of expression levels. Bioinformatics.
2012;28(8):1086–92.]
IDBA-Tran [Peng Y, Leung HC, Yiu SM, Lv MJ, Zhu XG, et al. IDBA-tran: a more robust
de novo de Bruijn graph assembler for transcriptomes with uneven
expression levels. Bioinformatics. 2013;29(13):i326–334.]