# Reference-Based Transcript Identification
- [Cufflinks](http://cole-trapnell-lab.github.io/cufflinks)
    - Trapnell, C., Williams, B. A., Pertea, G., Mortazavi, A., Kwan, G., van Baren, M. J., et al. (2010). Transcript assembly and quantification by RNA-Seq reveals unannotated transcripts and isoform switching during cell differentiation. [Nature Biotechnology, 28(5), 516–520](http://doi.org/10.1038/nbt.1621)
- [StringTie](https://ccb.jhu.edu/software/stringtie)
    - Pertea, M., Pertea, G. M., Antonescu, C. M., Chang, T.-C., Mendell, J. T., & Salzberg, S. L. (2015). StringTie enables improved reconstruction of a transcriptome from RNA-seq reads. [Nature Biotechnology, 33(3), 290–295](http://doi.org/10.1038/nbt.3122)
- [TransComb](https://sourceforge.net/projects/transcriptomeassembly/files)
    - Liu, J., Yu, T., Jiang, T., & Li, G. (2016). TransComb: genome-guided transcriptome assembly via combing junctions in splicing graphs. [Genome Biology, 17(1), 213](http://doi.org/10.1186/s13059-016-1074-1)
- [Scallop](https://github.com/Kingsford-Group/scallop)
    - Shao, M., & Kingsford, C. (n.d.). Accurate assembly of transcripts through phase-preserving graph decomposition. [Nature Biotechnology](http://doi.org/10.1038/nbt.4020)
- [Strawberry](https://github.com/ruolin/strawberry)
    - Liu, R., & Dickerson, J. (2017). Strawberry: Fast and accurate genome-guided transcript reconstruction and quantification from RNA-Seq. [PLoS Computational Biology, 13(11), e1005851–25](http://doi.org/10.1371/journal.pcbi.1005851)
Bayesember [Maretty L, Sibbesen JA, Krogh A. Bayesian transcriptome assembly. Genome
Biol. 2014;15(10):501.]
Scripture [Guttman M, Garber M, Levin JZ, Donaghey J, Robinson J, et al. Ab initio reconstruction of cell type-specific transcriptomes in mouse reveals the conserved multi-exonic structure of lincRNAs. Nat Biotechnol. 2010;28(5):503–10.]
IsoInfer [Feng J, Li W, Jiang T. Inference of isoforms from short sequence reads. J Comput Biol. 2011;18(3):305–21.]
IsoLasso [Li W, Feng J, Jiang T. IsoLasso: a LASSO regression approach to RNA-Seq based transcriptome assembly. J Comput Biol. 2011;18(11):1693–707.]
iReckon [Mezlini AM, Smith EJ, Fiume M, Buske O, Savich GL, et al. iReckon: simultaneous isoform discovery and abundance estimation from RNA-seq data. Genome Res. 2013;23(3):519–29.]
CEM [Li W, Jiang T. Transcriptome assembly and isoform expression level estimation from biased RNA-Seq reads. Bioinformatics. 2012;28(22):2914–21.]
Traph [Tomescu AI, Kuosmanen A, Rizzi R, Makinen V. A novel min-cost flow method for estimating transcript expression with RNA-Seq. BMC Bioinf. 2013;14 Suppl 5:S15.]
CIDANE [Canzar S, Andreotti S, Weese D, Reinert K, Klau GW. CIDANE: comprehensive isoform discovery and abundance estimation. Genome Biol. 2016;17:16.]
SLIDE (Li JJ, Jiang CR, Brown JB, Huang H, Bickel PJ. Sparse linear modeling of next-generation mRNA sequencing (RNA-Seq) data for isoform discovery and abundance estimation. Proc Natl Acad Sci USA. 2011; 108:19867–19872. [PubMed: 22135461])


Steijger T, Abril JF, Engstrom PG, Kokocinski F, Hubbard TJ, et al.
Assessment of transcript reconstruction methods for RNA-seq. Nat
Methods. 2013;10(12):1177–84.